chia-client/src/peer.rs

```diff
@@ -32,6 +32,40 @@ pub struct Peer {
     nonce: Mutex<u16>,
 }
 
+struct ResponseHandler {
+    requests: Requests,
+    message_id: u16,
+}
+
+impl ResponseHandler {
+    async fn new(
+        requests: Requests,
+        message_id: u16,
+    ) -> (ResponseHandler, oneshot::Receiver<Message>) {
+        let (sender, receiver) = oneshot::channel::<Message>();
+        requests.lock().await.insert(message_id, sender);
+
+        (
+            Self {
+                requests,
+                message_id,
+            },
+            receiver,
+        )
+    }
+}
+
+impl Drop for ResponseHandler {
+    fn drop(&mut self) {
+        let requests = Arc::clone(&self.requests);
+        let message_id = self.message_id;
+
+        tokio::spawn(async move {
```

Comment:
I'm not super familiar with `tokio`, but spawning a coroutine without storing a handle to it looks like a mistake. there won't be a way to synchronize/wait for this coroutine to exit.